### DISCUSSION TITLE
Azure Api Builder with docker .env

### DISCUSSION THREAD

**Discussion by Chr0no591** [Azure/data-api-builder#1830] [Q&A]
I'm currently trying to use a .env file together with the dab-config.json in a docker container.

Since Azure recently published the option to use environments. I changed the dab-config.json by using the @env() syntax for the credentials. The credentials are stored in a seperate .env file in the same folder like the dab-config.

My problem is, that when I run following command it is loading only the dab-config.json. I don't know how to reference the .env file with docker. The command looks as follows.

`sudo docker run -it -v ".:/App/samples" -p PORT:PORT mcr.microsoft.com/azure-databases/data-api-builder --ConfigFileName ./samples/dab-config.json`

I tried to use --env-file with the path to the env file. I tried to add another command --ConfigFileName with path to the env file
I was following this blog: https://devblogs.microsoft.com/azure-sql/dab-envfiles//#comment-210

**Comment by dbapibuilder** [comment]
If you are hosting the Web API on Azure App Service, you can setup like below

![image](https://github.com/Azure/data-api-builder/assets/148594852/0245c383-0608-4582-9f5b-670ef8e3c71d)

**Comment by yorek** [comment]
It should just work, make sure you use the `--env-file` option *before* specifying the image you want to use. For example:

```bash
docker run -it -v ".:/App/samples" --env-file .env -p 5000:5000 mcr.microsoft.com/azure-databases/data-api-builder --ConfigFileName ./samples/dab-config.json
```

Also, docker doesn't like quotes or double quotes in the .env files, so make sure to specify your connection string without enclosing quotes:

```
MSSQL=Server=myserver.database.windows.net;Initial Catalog=mydb;Persist Security Info=False;User ID=dab_user;Password=mypassword;MultipleActiveResultSets=False;Encrypt=True;Connection Timeout=30;
```

I just tested it both with Powershell on Windows and bash on WSL/Ubuntu and had no issues
